File: app/src/main/java/com/redkey/kotlintest/kotlin/design_pattern/抽象工厂模式/java/factory/ConcreteFactory2.java
package com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.factory;

import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.abst.ProductA;
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.abst.ProductB;
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.modle.ConcreteProductA2;
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.modle.ConcreteProductB2;

public class ConcreteFactory2 extends Factory {

    @Override
    public ProductA createProductA() {
        return new ConcreteProductA2();
    }

    @Override
    public ProductB createProductB() {
        return new ConcreteProductB2();
    }
}

File: app/src/main/java/com/redkey/kotlintest/kotlin/design_pattern/抽象工厂模式/kotlin/factory/ConcreteFactory2.kt
package com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.factory

import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.abst.ProductA
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.abst.ProductB
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.modle.ConcreteProductA2
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.modle.ConcreteProductB2

class ConcreteFactory2 : Factory() {
    override fun createProductA(): ProductA {
        return ConcreteProductA2()
    }

    override fun createProductB(): ProductB {
        return ConcreteProductB2()
    }
}